File: Bicicletas/src/Bicicletas/FactoriaCarretera.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Bicicletas;

/**
 *
 * @author ravolk
 */
public class FactoriaCarretera implements FactoriaCarreraYBicicleta{
    
    
    @Override
    public CarreraCarretera crearCarrera()
    {
        return new CarreraCarretera();
    }
    
    @Override 
    public BicicletaCarretera crearBicicleta(int identificador)
    {
        return new BicicletaCarretera(identificador);
    }
    
}

File: Bicicletas/src/Bicicletas/factoria_carretera.rb

require_relative 'FactoriaCarreraYBicicleta'
require_relative 'CarreraCarretera'
require_relative 'BicicletaCarretera'

class FactoriaCarretera < FactoriaCarreraYBicicleta

  def crearCarrera
    return CarreraCarretera.new
  end

  def crearBicicleta
    return BicicletaCarretera.new
  end

end
